filters/builtin/compress.go

```diff
@@ -219,11 +232,12 @@ func acceptedEncoding(r *http.Request) string {
 		}
 
 		name := strings.ToLower(strings.TrimSpace(sp[0]))
-		if !stringsContain(supportedEncodings, name) {
+		prio := index(c.encoding, name)
```

Comment:
If we are looking to have a priority for the same quality value (or absence thereof) then I think we should look at stable sort https://pkg.go.dev/sort#Stable that preserves client preferred algorithm order from the header.

Reply:
@AlexanderYastrebov clients tell they can do X,Y,Z and the priority gives the owner of skipper a choice what algorithm should be preferred. For example you might want to use brotli as first choice and fallback to gzip. A Browser sends in some random order what they support, and then you get something you are lucky and have no control.

Reply:
> A Browser sends in some random order what they support, and then you get something you are lucky and have no control.

Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding I think client sends its preferences using quality value for ordering (default quality value when absent is 1). The server serves the client so I think the logic should obey client preferences, i.e. use provided or default quality value without any additional server-side priority.

Reply:
@AlexanderYastrebov you are correct, the priority is kept on `quality value` if present but
checking on my Chrome browser, I can see the following:
`Accept-Encoding: gzip, deflate, br`

=> does it mean no preference or that `gzip` should be prioritised over `br`?

Reply:
I would say so, but I don't see order-dependent preference specified anywhere.
This answer https://stackoverflow.com/questions/3225136/http-what-is-the-preferred-accept-encoding-for-gzip-deflate also says first should be preferred but provides no reference.

If we would like to have a server side preference list then it would be interesting to address 
- why would brotli have priority over gzip (given that chrome does not do that and that it would be a change of behavior for the existing configurations)
- should the server-side priority list be configurable

E.g. Caddy implements configurable server side preference afaict from https://github.com/caddyserver/caddy/issues/3692

Reply:
I prefer the approach without the additional p field, and using stable sort to preserve the client specified order (which would be a fix of the current state in master)

Reply:
We've discussed this with @aryszka and came to conclusion that we should have configurable server-side priority for the algorithms in case quality values are equal (including default 1).

The default priority ordering should put brotly at the end (i.e. the priority should be `gzip > deflate > br`) to avoid unexpected algorithm change for the existing deployments.
Skipper operators then would be able to change the priority using a flag, something like `-compress-encoding-prefer=br,gzip,deflate` or e.g. exclude an algorithm like `-compress-encoding-prefer=br,gzip`.

Reply:
thanks a lot for your feedbacks :)
I would suggest `compress-encodings` in place of `compress-encoding-prefer` (shorter)